tests/test_import.py

```diff
@@ -218,3 +222,50 @@ def test_import_corrupted(fixtures):
                 .dicts()) == [
                     {'has_file': 'M', 'wants_file': 'Y'}
     ]
+
+
+def test_watchdog(fixtures):
+    """Checks that the file system observer triggers imports on new/changed files"""
+    tmpdir = fixtures['root']
+
+    acq_dir = tmpdir.join('12345678T000000Z_inst_zab')
+
+    node = st.StorageNode.get(st.StorageNode.name == 'x')
+
+    from watchdog.observers import Observer
+    obs = Observer()
+    obs.schedule(auto_import.RegisterFile(node),
+                 str(tmpdir),
+                 recursive=True)
+
+    obs.start()
+
+    f = acq_dir.join('new_file.log')
+    f.write('')
+
+    # this file is outside any acqs and should be ignored
+    g = tmpdir.join('some_file.log')
+    g.write('Where is my acq?!')
+
+    # nothing happens until the observer is running
+    assert file_copy_count('new_file.log') == 0
```

Comment:
These assertions seem a little fragile as they rely on the latency of the watchdog making the import call being longer than the time it takes from writing the file to getting to this point. Similarly if the latency was longer than expected (> 1s) the tests below might fail.